### PR TITLE
Fix for issue #567.

### PR DESCRIPTION
**Pull request by boaks**
Replacement of java.nio.charset.StandardCharsets, which requires java
1.7, to support older android versions.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: californium-core/src/test/java/org/eclipse/californium/core/test/RandomAccessBlockTest.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public void testServerReturnsIndividualBlocks() throws Exception {
 		private BlockwiseResource(String name, String responsePayload) {
 			super(name);
 			this.responsePayload = responsePayload;
-			buf = ByteBuffer.wrap(responsePayload.getBytes(StandardCharsets.US_ASCII));
+			buf = ByteBuffer.wrap(responsePayload.getBytes(CoAP.UTF8_CHARSET));
```
Comment (Thomas-Vos) [inline californium-core/src/test/java/org/eclipse/californium/core/test/RandomAccessBlockTest.java]: Is this supposed to be a different charset or is this a typo?

**Comment by sbernard31** [comment]
No objection, I just ask myself which android API are we focusing ?

I understand that this fix will help us to support API 16.
If I well understand before this fix we was API 19 compliant.

I try to look if there is maven plugin to ensure android API level compatibility but I didn't find anything good... (just this old (unmaintained?) project : https://code.google.com/archive/p/compliancechecker/)

Here is [Android_version_history](https://en.wikipedia.org/wiki/Android_version_history)
And here [some statistics](https://developer.android.com/about/dashboards/index.html#Platform) about uses of android version. (This masked some devices as it only counts devices connected to googleplay store)

I suppose the idea for now about android compliance is just best effort driving by community needs.

**Comment by boaks** [comment]
I changed the minSdkVersion from 15 to 19 with commit 0d3a6f9d4e727d760d564b8512d6086dee0b3c81 3 months ago, 
mainly for the `System.lineSeparator()`.

I don't plan to really support older versions, but, if someone else feels it works with small changes, that's OK for me. Therefore I created this PR.

**Comment by sbernard31** [comment]
> if someone else feels it works with small changes, that's OK for me

That's ok for me too. 

I forgot we have an android demo. Should create a build about it to ensure that we are compatible with a given version of android API ?
I try to build it but we need a very recent version of gradle (on jenkins there is only the v3.1)
Maybe using gradle wraper ? (I'm not an expert about Gradle)

**Comment by boaks** [comment]
Do you think on a static compatibility check?
I would not spend time on that. 
The most issues of the last half year only appeared at runtime.

**Comment by sbernard31** [comment]
I rather continue the discussion in #567

**Comment by boaks** [comment]
Mark the TLS and File NetworkConfig to be excluded from the check.
Android developer may chose to configure by code or use the AssetManager / InputStream variants.

**Comment by sbernard31** [comment]
LGTM !
(It was hard to search "real" changes between all the big  `System.lineSeparator()`/`StringUtil.lineSeparator()`  refactoring ;-) )

Just one open question : 
Using @NotForAndroid to ignore compatibility check is great ! :+1: 
Should we also use it as user documentation ? if yes, I think we should also mark `createWithFile`, `createStandardWithFile` too.

**Comment by boaks** [comment]
> Should we also use it as user documentation ? if yes, I think we should also mark createWithFile, createStandardWithFile too.

Yes, your right.
I also forgot to make the "toString(InetSocketAddress)" check, if the getHostString is available.

**Comment by boaks** [comment]
I tested it successful with API 19 and 23.
So we will see, if it works with older API version or if we get new error issues :-).

**Pull request merged**